### PR TITLE
Auto: implement issue #142

### LINKED ISSUE #142: [Test] Add unit tests for AuthService — generate/verify/refresh_token
## Task
Create `tests/unit/test_auth_service.py` covering JWT token operations in `AuthService`.

## Implementation
1. Fixture `auth_service` that sets `JWT_SECRET_KEY` env var and instantiates `AuthService(secret_key=...)`
2. Tests:
   - `test_generate_token_returns_string`
   - `test_generate_token_without_tenant`
   - `test_verify_token_valid`
   - `test_verify_token_invalid`
   - `test_verify_token_tampered` (different secret)
   - `test_refresh_token_valid`
   - `test_refresh_token_invalid`
   - `test_is_valid_email`

## Verification
- `pytest tests/unit/test_auth_service.py --collect-only` — >= 7 items, 0 errors
- `pytest tests/unit/test_auth_service.py -v` — all pass

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #142

Plan: `.plans/issue-142.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-142.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for authentication service JWT token operations and email validation, covering valid scenarios, edge cases, and error handling.

[Review Change Stack](https://app.coderabbit.ai/change-stack/yihua-zhuo/agent-job/pull/439)

### CODE REVIEW

File: tests/unit/test_auth_service.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+class TestRefreshToken:
+    """Tests for AuthService.refresh_token."""
+
+    @pytest.mark.asyncio
+    async def test_refresh_token_valid(self, auth_service):
+        """refresh_token should return a new non-empty string different from the original."""
+        original = auth_service.generate_token(1, "alice", "admin", 10)
+        time.sleep(1.1)
```
Comment (yihua-zhuo) [inline tests/unit/test_auth_service.py:66]: **WARNING** (suggestion)

**Issue:** time.sleep(1.1) slows the test suite and makes it depend on real clock passage instead of mocking time (Rule 63, 35)

**Suggestion:** Use freezegun or monkeypatch datetime to advance time deterministically, removing the 1.1s sleep

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `pass`
**Inline comments posted:** 1

**Summary:** Reviewed 2 file(s) between origin/master and origin/auto/issue-142.

### Issues without a usable file/line
- `.plans/issue-142.md`: Implementation plan is clear and well-scoped; no issues found.
